__tests__/app.test.js

```diff
@@ -199,4 +199,100 @@ describe('GET api/articles/:article_id/comments', () => {
+                const comment = body.comment
+                expect(comment).toMatchObject({
+                    "comment_id" : 19,
+                    "votes" : 0,
+                    "created_at" : expect.any(String),
+                    "author" : "rogersop",
+                    "body" : "spaceships dont come equipped with rear view mirrors",
+                    "article_id" : 1
+                })
+            })          
+    })
+    test(`sending a POST request to api/articles/:articles_id/comments with a username found within the users table, but an article_id that doesnt exist returns a 400 with message "bad request".`, () => {
+        return request(app)
+            .post('/api/articles/5000/comments')
+            .send({ 
+                username: "rogersop", 
+                body : "be responsible for what you say because the words you speak can truly cause dismay"
+            })
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe("bad request")
+            })          
+    })
+    test(`sending a POST request to api/articles/:articles_id/comments with a username found within the users table, but a completely invalid article_id (not a number) returns a 400 with message "bad request".`, () => {
+        return request(app)
+            .post('/api/articles/andrewdoesntsweat/comments')
+            .send({ 
+                username: "rogersop", 
+                body : "america america now im going africa my death is money"
+            })
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe("bad request")
+            })          
+    })
+    test(`sending a POST request to api/articles/:articles_id/comments with a username not found within the users table sends back a 400 status with the message - "bad request".`, () => {
+        return request(app)
+            .post('/api/articles/1/comments')
+            .send({ 
+                username: "GiveDollarRobert", 
+                body : "I close my eyes and seize it, i clench my fists and beat it, i light my torch and burn it, i am the beast i worship"
+            })
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe("bad request")
+            })          
+    })
+    test(`sending a POST request to api/articles/:articles_id/comments with a valid username ONLY returns an error code 400 and a msg "bad request".`, () => {
+        return request(app)
+            .post('/api/articles/1/comments')
+            .send({ 
+                username: "rogersop" 
+            })
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe("bad request")
+            })          
+    })
+    test(`sending a POST request to api/articles/:articles_id/comments with a valid username, a valid body AND an extra key returns an error code 400 with message "bad request".`, () => {
```

Comment:
So I think for this one, we should set up our controllers/models so that as long as we have a username and a body the request can be made and any extra properties are ignored.